frontend/config-assistant/src/helpers/schema/schemaPreprocessor.ts

```diff
@@ -21,44 +21,73 @@ const preprocessedRefSchemas: Map<string, JsonSchemaObjectType> = new Map();
  * - Merges back oneOfs and anyOfs with just one single entry into the schema
  * - If it is possible: merges oneOfs into anyOfs
  *
- * @param schema the schema to preprocess
+ * @param schema2 the schema to preprocess
  * @returns the preprocessed schema
  */
-export function preprocessSchema(schema: JsonSchemaObjectType): JsonSchemaObjectType {
+export function preprocessSchema(schema2: JsonSchemaType): JsonSchemaType {
+  if (typeof schema2 !== 'object') {
```

Comment:
copying the schema is still necessary because of the reference resolving (and maybe other things that modify the schema)